### PR TITLE
feat: Use string as default for unknown column type

### PR DESCRIPTION
**Pull request by LeeMendelowitz**
PR for Issue https://github.com/r-dbi/RMariaDB/issues/259.

Querying a JSON column on MySQL 8.0.26 returns a column of type 245, which is not handled by RMariaDB. This PR converts column of unknown type to string and issues a warning, which is similar to the RMySQL behavior.

### CODE REVIEW

**Comment by LeeMendelowitz** [comment]
@[krlmlr](https://github.com/krlmlr) I'm curious to hear what you think about this proposed solution whenever you have a chance.

**Review by krlmlr — APPROVED** [review]
Thanks, looks great!

File: tests/testthat/test-dbWriteTable.R [hunk 1]
```diff
@@ -79,7 +79,12 @@ test_that("writing and reading JSON (#127)", {
   x <- data.frame(col1 = "[1,2,3]", stringsAsFactors = FALSE)
 
   dbWriteTable(con, "t1", x, field.types = c(col1 = "json"), overwrite = TRUE, temporary = TRUE)
-  dbReadTable(con, "t1")
 
-  expect_equal(dbReadTable(con, "t1"), x)
+  d <- dbReadTable(con, "t1")
```
[the remark this answers is not in the document]
Comment (krlmlr) [reply tests/testthat/test-dbWriteTable.R]: Do you see why the warning doesn't seem to be raised here?

**Comment by MichaelSchatz** [comment]
@krlmlr This would be a really helpful feature to have integrated into the package for a project I'm currently working on. Is there any chance of getting it merged into the next release candidate?

File: src/MariaTypes.cpp [hunk 2]
```diff
@@ -51,7 +52,8 @@ MariaFieldType variable_type_from_field_type(enum_field_types type, bool binary,
   case MYSQL_TYPE_NULL:
     return MY_INT32;
   default:
-    throw std::runtime_error("Unimplemented MAX_NO_FIELD_TYPES");
+    warning("unrecognized field type %i imported as character", type);
```
Comment (krlmlr) [inline src/MariaTypes.cpp:55]:
```suggestion
    cpp11::warning("unrecognized field type %i imported as character", type);
```

**Comment by krlmlr** [comment]
@LeeMendelowitz: warnings don't seem to be raised for MariaDB. What do you see when you run the following:

``` r
DBI::dbGetInfo(RMariaDB::mariadbDefault())$db.version
#> [1] "10.10.2-MariaDB"
```

<sup>Created on 2023-03-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

? I'd like to stabilize the tests before merging.

**Comment by krlmlr** [comment]
I added MySQL 8.0 configurations to main and merged it here, let's see what the tests say.

@LeeMendelowitz: can you please help by reporting what the following code shows on your system:

``` r
DBI::dbGetInfo(RMariaDB::mariadbDefault())$db.version
#> [1] "10.10.2-MariaDB"
```

**Comment by LeeMendelowitz** [comment]
@krlmlr Sorry for the delay. I get "5.5.5-10.11.2-MariaDB-1:10.11.2+maria~ubu2204".

For some reason `RMariaDB::mariadbDefault()`was not working for me when running off of the `LeeMendelowitz:mysql8_json_support` branch. I would get:

```
RMariaDB::mariadbDefault()
Error: Reason: Test database not available
```

It didn't seem to be using the `[rs-dbi]` group in my `~/.my.cnf`, so I explicitly provided username + password + host + port instead.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**